odl/solvers/util/callback.py

```diff
@@ -25,6 +27,62 @@
            'CallbackProgressBar')
 
 
+def call_callback(callback, x, **kwargs):
+    """Call a callback with optional arguments.
+
+    Parameters
+    ----------
+    callback : callable or None
+        Function taking at least one argument. If `None`, does nothing.
+    x :
+        The main object to call the callback with, typically the current
+        iterate.
+    kwargs :
+        Other parameters that should be called. These are only used if the
+        callback accepts them.
+
+    Examples
+    --------
+    Simple usage:
+
+    >>> def callback(x):
+    ...     print(x)
+    >>> call_callback(callback, 'arg')
+    arg
+
+    Optionally ignore one argument:
+
+    >>> call_callback(callback, 'arg', b='other_arg')
+    arg
+    >>> def other_callback(x, b=None):
+    ...     print(x, b)
+    >>> call_callback(other_callback, 'arg', b='other_arg')
+    arg other_arg
```

Comment:
The example is  really confusing. Above it says "Optionally ignore one argument", but the example doesn't do that, it just happens to print nothing for the default. Nowhere is `None` ignored. I was looking in the code below for something that ignores `None`, without success.

Reply:
I also find using strings and the name `'arg'` confusing, since it looks as if `'arg'` was the name of a parameter, where it's actually just some value. How about just using numbers in the example?